src/documentation/webview/webview.ts

```diff
@@ -0,0 +1,123 @@
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VS Code Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import { RenderNode, WebviewContent, WebviewMessage } from "./WebviewMessage";
+import { createCommunicationBridge } from "./CommunicationBridge";
+
+createCommunicationBridge().then(async bridge => {
+    const vscode = acquireVsCodeApi();
+    let activeDocumentationPath: string | undefined;
+    let contentToApplyOnRender: RenderNode | undefined;
+
+    window.document.getElementById("app");
+
+    // Handle messages coming from swift-docc-render
+    bridge.onDidReceiveMessage(message => {
+        switch (message.type) {
+            case "rendered":
+                if (contentToApplyOnRender) {
+                    setTimeout(() => {
+                        bridge.send({ type: "contentUpdate", data: contentToApplyOnRender });
+                        contentToApplyOnRender = undefined;
+                    }, 1);
+                    break;
+                }
+                vscode.postMessage({ type: "rendered" });
+                break;
+        }
+    });
+
+    // Handle messages coming from vscode-swift
+    window.addEventListener("message", event => {
+        if (typeof event.data !== "object" || !("type" in event.data)) {
+            return;
+        }
+
+        const message = event.data as WebviewMessage;
+        switch (message.type) {
+            case "update-content":
+                handleUpdateContentMessage(message.content);
+                break;
+        }
+    });
+    function handleUpdateContentMessage(content: WebviewContent) {
+        if (content.type === "render-node") {
+            const renderNode = content.renderNode;
+            const documentationPath: string = (() => {
+                switch (renderNode.kind) {
+                    case "symbol":
+                    case "article":
+                        return "/live/documentation";
+                    case "overview":
+                        return "/live/tutorials-overview";
+                    default:
+                        return "/live/tutorials";
+                }
+            })();
+            if (activeDocumentationPath !== documentationPath) {
+                activeDocumentationPath = documentationPath;
+                contentToApplyOnRender = renderNode;
+                bridge.send({
+                    type: "navigation",
+                    data: documentationPath,
+                });
+            } else {
+                bridge.send({ type: "contentUpdate", data: renderNode });
+            }
+        } else {
+            // Show the error message
+        }
+    }
+
+    // Notify vscode-swift that we're ready to receive messages
+    vscode.postMessage({ type: "loaded" });
+});
+
+declare global {
```

Comment:
other webviews we do could use this so we could move it to a common place

Reply:
I agree, but I think we should wait until we actually have another webview first.